File: 100dynamic/746.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp0, dp1 = 0,0

        for i in range(2, len(cost)+1):
            dp = min(dp0 + cost[i-2], dp1 + cost[i-1])
            dp0 = dp1
            dp1 = dp
        return dp1
        

File: 100dynamic/746.rs
use std::cmp;
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut dp0: i32 = 0;
        let mut dp1: i32 = 0;
        let mut dp: i32 = 0;

        for i in 2..cost.len()+1 {
            dp = cmp::min(dp0 + cost[i as usize - 2], dp1 + cost[i as usize - 1]);
            dp0 = dp1;
            dp1 = dp;
        }
        dp1
    }
}